twenty-first/src/shared_math/x_field_element.rs

```diff
@@ -305,7 +304,22 @@ impl Add for XFieldElement {
 
 */
 
-impl Mul for XFieldElement {
+impl Mul<BFieldElement> for XFieldElement {
```

Comment:
```suggestion
/// Scalar multiplication of `XFieldElement`
impl Mul<BFieldElement> for XFieldElement {
```

Reply:
I disagree with this comment. I'm not sure exactly what "scalar" refers to.